$NetBSD$

* Deal with a failure specific to Clang.
* Work around missing fseeko64() et al.

--- CMakeLists.txt.orig	2023-09-09 06:14:53.000000000 -0700
+++ CMakeLists.txt	2023-09-11 19:54:24.624246614 -0700
@@ -356,6 +356,7 @@
 		add_definitions(-Wno-nullability-completeness)
 		add_definitions(-Wno-tautological-pointer-compare)
 		add_definitions(-Wno-deprecated-register)
+		add_definitions(-Wno-c++11-narrowing)
 	endif()
 
 	if(USE_ASAN)
@@ -425,7 +426,9 @@
 		endif()
 		add_definitions(-D_XOPEN_SOURCE=700)
 		add_definitions(-D_XOPEN_SOURCE_EXTENDED -D__BSD_VISIBLE=1 -D_BSD_SOURCE -D_DEFAULT_SOURCE)
+		if(NOT ${CMAKE_SYSTEM_NAME} STREQUAL "NetBSD")
 		add_definitions(-D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64)
+		endif()
 		if(${CMAKE_SYSTEM_NAME} STREQUAL "NetBSD")
 			add_definitions(-D_NETBSD_SOURCE)
 		endif()
